eventdata/parameter_sources/interval_query_source.py

```diff
@@ -0,0 +1,136 @@
+        else:
+            raise ConfigurationError("Parameter 'body' must be specified.");
+
+        if 'min_interval_size_pct' in params.keys():
+            self._min_interval_size_pct = float(params['min_interval_size_pct']);
+            if self._min_interval_size_pct <= 0:
+                raise ConfigurationError("Parameter 'min_interval_size_pct' must be > 0.");
+            if self._min_interval_size_pct >= 100:
+                raise ConfigurationError("Parameter 'min_interval_size_pct' must be < 100.");
+        else:
+            raise ConfigurationError("Parameter 'min_interval_size_pct' must be specified.");
+
+        if 'max_interval_size_pct' in params.keys():
+            self._max_interval_size_pct = float(params['max_interval_size_pct']);
+            if self._max_interval_size_pct <= self._min_interval_size_pct:
+                raise ConfigurationError("Parameter 'max_interval_size_pct' must be > 'min_interval_size_pct'.");
+            if self._max_interval_size_pct > 100:
+                raise ConfigurationError("Parameter 'max_interval_size_pct' must be <= 100.");
+        else:
+            raise ConfigurationError("Parameter 'max_interval_size_pct' must be specified.");
+
+    def partition(self, partition_index, total_partitions):
+        return self
+
+    def size(self):
+        return 1
+
+    def params(self):
+        key = "{}_{}".format(self._index_pattern, self._fieldname);
+        if key in fs.global_fieldstats.keys():
+            stats = fs.global_fieldstats[key];
+        else:
+            raise ParameterSourceError("No statistics found for field `{}` in index pattern `{}`.".format(self._fieldname, self._index_pattern));
+
+        range_max = stats['max'];
+        range_min_upper = int(stats['max'] - ((stats['max'] - stats['min']) * self._min_interval_size_pct / 100.0));
+        delta = (stats['max'] - stats['min']) * ((self._max_interval_size_pct - self._min_interval_size_pct) / 100.0);
+        range_min = range_min_upper - int(delta * random.random());
+        body = copy.deepcopy(self._body);
+
+        # Rewrite query to include new range filter based on range_min and range_max.
+        # This assumes that the body contains `query.bool.must` key.
+        if not isinstance(body['query']['bool']['must'], list):
+            body['query']['bool']['must'] = [ body['query']['bool']['must'] ];
+
+        range_clause = {}
+        range_clause['range'] = {}
+        range_clause['range'][self._fieldname] = { 'gte': range_min, 'lte': range_max, 'format': 'epoch_millis'}
+        body['query']['bool']['must'].append(range_clause);        
+
+        logger.info("[interval_query] Interval generated for field `{}`: Min: {} [{}, {}], Max: {}".format(self._fieldname, range_min, stats['min'], range_min_upper, range_max));
+ 
+        request = { 'index': self._index_pattern, 'type': self._type, 'body': body, 'use_request_cache': self._cache};
```

Comment:
the parameter `use_request_cache` is incompatible with recent Rally versions. It needs to be called `cache` now. See http://esrally.readthedocs.io/en/stable/migrate.html#custom-parameter-sources for details.

Reply:
Will do.